### PR TITLE
fix: line chart wont interpolate on data changes

### PR DESCRIPTION
**Pull request by Amurmurmur**
After upgrading to Expo v49 and react-native-reanimated v3 the chart will not interpolate on data switch due to `previous` value always being returned as null. 

The same code works fine with Reanimated v2 and expo v48, most likely a buggy on the reanimated side (issue raised with them no feedback yet)

As of now this PR fixes it to support Reanimated v3 and Expo v49.

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -13,26 +13,29 @@
     "postinstall": "patch-package"
   },
   "dependencies": {
+    "@shopify/react-native-skia": "0.1.196",
```
Comment (cam-shaw) [inline example/package.json:16]: Is there a reason you added skia to the project?

Comment (cam-shaw) [inline example/package.json:16]: Is there a reason you added skia to the project?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-native-wagmi-charts",
-  "version": "2.3.0",
```
Comment (cam-shaw) [inline package.json:3]: Revert this change as the version bump would happen when a release is cut.

File: example/package.json [hunk 3]
```diff
@@ -13,26 +13,29 @@
...
-    "react-native-gesture-handler": "~2.9.0",
-    "react-native-reanimated": "~2.14.4",
-    "react-native-svg": "13.4.0",
-    "react-native-web": "~0.18.11"
+    "react-native": "0.72.3",
+    "react-native-gesture-handler": "~2.12.0",
+    "react-native-reanimated": "^3.3.0",
+    "react-native-svg": "13.9.0",
+    "react-native-web": "~0.19.6",
+    "typescript": "^5.1.3"
```
Comment (cam-shaw) [inline example/package.json:30]: It might be good to break all of the expo 49 and expo package updates into a separate PR so changes can be isolated. Quite a lot has been updated here. How much of is it related to the reanimated 3.x upgrade?

**Review by cam-shaw — CHANGES_REQUESTED** [review]
As this PR is for a line interpolation fix, it would be good to isolate the changes to just what's needed to fix the bug. 
Great to see the contribution as it's a good one for the project! 🙌

**Comment by LeoAnt02** [comment]
Hey @Amurmurmur ,
Thanks so much for this PR

Just curious, how is this tested on web ? 
Expo SDK49 introduce the metro web bundler that will slowly replace webpack 

I think we should ensure that we support both
